metrics/core/test/util/validation/printer/validation_result_printer_test.dart

```diff
@@ -0,0 +1,311 @@
+          target: target,
+          conclusion: conclusion,
+          description: description,
+        );
+        final results = {target: result};
+        final validationResult = ValidationResult(results);
+
+        resultPrinter.print(validationResult);
+
+        verify(
+          ioSink.writeln(argThat(contains(description))),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".print() prints the validation result message to the sink with the given validation result details",
+      () {
+        const details = {'test': 'details'};
+        const result = TargetValidationResult(
+          target: target,
+          conclusion: conclusion,
+          details: details,
+        );
+        final results = {target: result};
+        final validationResult = ValidationResult(results);
+
+        resultPrinter.print(validationResult);
+
+        verify(
+          ioSink.writeln(argThat(containsAllEntries(details))),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".print() prints the validation result message to the sink with the given validation result context",
+      () {
+        const context = {'test': 'context'};
+        const result = TargetValidationResult(
+          target: target,
+          conclusion: conclusion,
+          context: context,
+        );
+        final results = {target: result};
+        final validationResult = ValidationResult(results);
+
+        resultPrinter.print(validationResult);
+
+        verify(
+          ioSink.writeln(argThat(containsAllEntries(context))),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".print() prints validation result messages for all targets",
+      () {
+        final results = {
+          target: firstResult,
+          secondTarget: secondResult,
+          thirdTarget: thirdResult,
+        };
+        final validationResult = ValidationResult(results);
+
+        resultPrinter.print(validationResult);
+
+        results.forEach((target, result) {
+          verify(
+            ioSink.writeln(argThat(
+              validationResultMessageMatcher(target, result),
+            )),
+          ).called(once);
+        });
+      },
+    );
+
+    test(
+      ".print() prints the validation results in the given order",
+      () {
+        final results = {
+          target: firstResult,
+          secondTarget: secondResult,
+          thirdTarget: thirdResult,
+        };
+        final validationResult = ValidationResult(results);
+
+        resultPrinter.print(validationResult);
+
+        final expectedMessages = results.entries.map((entry) {
+          final target = entry.key;
+          final result = entry.value;
+
+          return argThat(
+            validationResultMessageMatcher(target, result),
+          );
+        });
+
+        verifyInOrder(
+          expectedMessages.map(ioSink.writeln).toList(),
+        );
+      },
+    );
+  });
+}
+
+class IOSinkMock extends Mock implements IOSink {}
```

Comment:
✅ 
Should we mock the `StringSink` directly to avoid confusion? Moreover, I suppose we can use [`StringBuffer`](https://api.dart.dev/stable/2.13.4/dart-core/StringBuffer-class.html) and don't use mock at all and then use the `stringContainsInOrder` on `buffer.toString()` where necessary.